fixrightclick.lua

```diff
@@ -0,0 +1,37 @@
+--@ module = true
+
+local overlay = require('plugins.overlay')
+
+RightClickWidget = defclass(RightClickWidget, overlay.OverlayWidget)
+RightClickWidget.ATTRS{
+    desc='When painting a rectangle, makes right click cancel selection instead of exiting.',
+    default_enabled=true,
+    viewscreens={
+        'dwarfmode/Building/Placement',
+        'dwarfmode/Designate',
+        'dwarfmode/Stockpile/Paint',
+        'dwarfmode/Zone/Paint',
+        'dwarfmode/Burrow/Paint'
+        },
+}
+
+local selection_rect = df.global.selection_rect
+local buildreq = df.global.buildreq
+
+function RightClickWidget:onInput(keys)
+    if keys._MOUSE_R or keys.LEAVESCREEN then
+        -- building mode, do not run if buildingplan.planner is enabled since it already provides this functionality
+        if buildreq.selection_pos.x >= 0 and not overlay.get_state().config['buildingplan.planner'].enabled then

```

Comment:
does it work fine if you just don't check for `buildingplan`? I don't think it matters which tool resets `selection_pos`

Reply:
that would be fine, yeah.